### PR TITLE
Edit an mq in trs

### PR DESCRIPTION
**Pull request by hortha**
### Context

As a Support team user
I want to be able to edit a MQ award in TRS
So that I can make sure that the teaching record has the correct award allocated.

### Changes proposed in this pull request

UI to edit Provider, Specialism, Start Date or Result of an MQ + supporting backend code and tests

### Guidance to review

CRM queries + UI + tests

### Checklist

-   [x] Attach to Trello card
-   [x] Rebased master
-   [ ] Cleaned commit history
-   [x] Tested by running locally

### CODE REVIEW

**Comment by hortha** [comment]
[Edit an MQ in TRS](https://trello.com/c/gDUIYLCJ/380-edit-an-mq-in-trs)

File: TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/QueryHandlers/GetQualificationByIdHandler.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+using TeachingRecordSystem.Core.Dqt.Queries;
+
+namespace TeachingRecordSystem.Core.Dqt.QueryHandlers;
+
+public class GetQualificationByIdHandler : ICrmQueryHandler<GetQualificationByIdQuery, dfeta_qualification?>
+{
+    public async Task<dfeta_qualification?> Execute(GetQualificationByIdQuery query, IOrganizationServiceAsync organizationService)
+    {
+        var filter = new FilterExpression();
+        filter.AddCondition(dfeta_qualification.PrimaryIdAttribute, ConditionOperator.Equal, query.QualificationId);
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/QueryHandlers/GetQualificationByIdHandler.cs:13]: We probably want an `Active` filter on this too

**Review by gunndabad — APPROVED** [review]

File: TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/QueryHandlers/GetQualificationByIdHandler.cs (hunk 1 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/QueryHandlers/GetQualificationByIdHandler.cs:13]: Might we have a future need to be able to toggle the status for a qualification in the TRS Console UI i.e. be able to show inactive quals too potentially?

Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/QueryHandlers/GetQualificationByIdHandler.cs:13]: Done

**Pull request merged**